src/components/profile/userContext.js

```diff
@@ -0,0 +1,388 @@
+          });
+      }
+    });
+
+  const setAuthorizationHeader = token => {
+    const Token = `Bearer ${token}`;
+    localStorage.setItem('Token', Token);
+    axios.defaults.headers.common['Authorization'] = Token;
+  };
+
+  const setUserHandleHeader = handle => {
+    localStorage.setItem('Handle', handle);
+  };
+
+  const getAuthenticated = () =>
+    new Promise(async (resolve, reject) => {
+      if (
+        localStorage?.Token &&
+        localStorage?.Handle &&
+        !isLoadingAuthenticated &&
+        !isAuthenticated
+      ) {
+        setIsLoadingAuthenticated(true);
+        const decodedToken = jwtDecode(localStorage?.Token);
+        if (decodedToken?.exp * 1000 < Date.now()) {
+          localStorage.removeItem('Token');
+          localStorage.removeItem('Handle');
+          delete axios.defaults.headers.common['Authorization'];
+          setisAuthenticated(false);
+          history.push('/login');
+        } else {
+          axios.defaults.headers.common['Authorization'] = localStorage?.Token;
+          await getCurrentUserData(localStorage?.Handle)
+            .then(() => {
+              setisAuthenticated(true);
+              resolve(currentUser);
+            })
+            .catch(err => {
+              setUserError(err);
+              reject(err);
+            })
+            .finally(() => {
+              setIsLoadingAuthenticated(false);
+            });
+        }
+      }
+    });
+
+  const logoutUser = () => {
+    localStorage.removeItem('Token');
+    localStorage.removeItem('Handle');
+    delete axios.defaults.headers.common['Authorization'];
+    setisAuthenticated(false);
+  };
+  // Passing state to value to be passed to provider
+  const value = {
+    isAuthenticated,
+    isLoadingAuthenticated,
+    getAuthenticated,
+    userList,
+    currentUser,
+    userError,
+    refreshUserList,
+    getCurrentUserData,
+    isLoadingUserList,
+    setIsLoadingUserList,
+    isLoadingUserData,
+    loginUser,
+    logoutUser,
+    isLoadingLogin,
+    lastRefreshUserList,
+    registerUser,
+    isLoadingRegister,
+    editUserDetail,
+    isLoadingEditUserDetail,
+    uploadImage,
+  };
+  return <userContext.Provider value={value}>{children}</userContext.Provider>;
+};
+
+/** A hook for consuming our User context in a safe way
+ * @example //getting the user list
+ * import { useUserData } from 'userContext'
+ * const { userList } = useUserData();
+ * @returns {User[]}
+ */
+export const useUserListData = () => {
+  // Destructuring value from provider
+  const ctx = useContext(userContext);
+
+  if (ctx === undefined) {
+    throw new Error('useUserListData must be used within a UserProvider');
+  }
+
+  const {
+    userList,
+    userError,
+    refreshUserList,
+    isLoadingUserList,
+    lastRefreshUserList,
+  } = ctx;
+
+  if (
+    !isLoadingUserList &&
+    (_.keys(userList).length === 0 ||
+      lastRefreshUserList === null ||
+      lastRefreshUserList >= Date.now + 600)
```

Comment:
I would require it to wait for a refresh even if the userlist is empty so it can't spam calls

Reply:
Fixed